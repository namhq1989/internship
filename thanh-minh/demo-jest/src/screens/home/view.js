import React from 'react'
import { connect } from 'dva'
import { Row, Col, Button } from 'antd'
import { translate } from 'react-i18next'
import { ImageConst } from '../../configs'
import './style.less'

class HomePageView extends React.Component {
  componentDidMount() {
    this.init()
    this.saveUserInfo()
  }

  init = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/init',
    })
  }

  saveUserInfo = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/userInfoSave',
    })
  }

  handleLogOut = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/logout',
    })
  }

  render() {
    const { home: { userInfo } } = this.props
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div>
        <Row type="flex" justify="center">
          <Col span={8} className="wrapper">
            <h1>{t('welcome-page')} <strong className="user-info">{userInfo.name}</strong>{t('come-page')}</h1>
            <Button
              type="primary"
              onClick={this.handleLogOut}
              className="logout-button"
            >{t('log-out')}
            </Button><br /><br />
            <div className="country-flag">
              <span>{t('language- label')}:&nbsp;&nbsp;</span>
              <Button onClick={() => changeLanguage('vi')}>
                <img src={ImageConst.imageIconVietNamese} alt="vn" />
              </Button>  &nbsp;&nbsp;&nbsp;&nbsp;
              <Button onClick={() => changeLanguage('en')} className="en-flae">
                <img src={ImageConst.imageIconEngland} alt="en" />
              </Button>
            </div>
          </Col>
        </Row>
      </div>
    )
  }
}

export default connect(({ home }) => ({ home }))(translate('translations')(HomePageView))
