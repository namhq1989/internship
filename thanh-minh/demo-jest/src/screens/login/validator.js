import { AppConst } from '../../configs'

const loginValidator = (email, password) => {
  if (!email || !password) {
    return { status: false, message: 'RequireEmailAndPassword' }
  } else if (!AppConst.regex.email.test(email)) {
    return { status: false, message: 'EmailIsNotValid' }
  } else if (password.length < 6) {
    return { status: false, message: 'PasswordLengthMustAtLeast6Characters' }
  } else {
    return { status: true, message: '' }
  }
}

export {
  loginValidator
}
