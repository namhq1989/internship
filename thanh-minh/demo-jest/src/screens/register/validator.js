import { AppConst } from '../../configs'

const registerValidator = (name, email, password, rePassword, age) => {
  if (!name || !email || !password || !rePassword || !age) {
    return { status: false, message: 'RequireRegister' }
  } else if (!AppConst.regex.email.test(email)) {
    return { status: false, message: 'EmailIsNotValid' }
  } else if (name.length < 2 || name.length > 128) {
    return { status: false, message: 'userNameLengthBetweenFrom2To128Characters' }
  } else if (password.length < 6) {
    return { status: false, message: 'PasswordLengthMustAtLeast6Characters' }
  } else if (rePassword !== password) {
    return { status: false, message: 'ComfirmToFirstPassword' }
  } else if (!age.match(AppConst.regex.ageInt)) {
    return { status: false, message: 'AgeIntegerNumber' }
  } else if (age < 1 || age > 100) {
    return { status: false, message: 'AgeMinMax' }
  } else {
    return { status: true, message: '' }
  }
}
export {
  registerValidator
}
