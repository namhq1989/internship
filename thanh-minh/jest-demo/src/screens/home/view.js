import React from 'react'
import { connect } from 'dva'
import { Row, Col, Button } from 'antd'
import { translate } from 'react-i18next'
import { LanguageChange } from '../../components'
import { KeysConst } from '../../configs/locale'
import './style.less'

export class HomePageView extends React.Component {
  componentDidMount() {
    this.init()
    this.saveUserInfo()
  }

  /**
   * Call to model for Check whether the user account is logged on
   */
  init = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/init',
    })
  }

  /**
   * Call to model for save user account into store
   */
  saveUserInfo = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/userInfoSave',
    })
  }

  /**
   * Call to model for log out
   */
  handleLogOut = () => {
    const { dispatch } = this.props
    dispatch({
      type: 'home/logout',
    })
  }

  /**
   * Render register view
   */
  render() {
    const { t, i18n, userInfo } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div>
        <Row type="flex" justify="center">
          <Col span={8} className="wrapper">
            <h1>{t(KeysConst.welcomePage)}
              <strong className="user-info">
                {userInfo.name}
              </strong>
              {t(KeysConst.comePage)}
            </h1>
            <Button
              type="primary"
              onClick={this.handleLogOut}
              className="logout-button"
            >{t(KeysConst.logOut)}
            </Button><br /><br />
            <LanguageChange
              t={t}
              changeLanguage={changeLanguage}
            />
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect(
  ({ home }) => (
    { userInfo: home.userInfo }
  )
)(translate('translations')(HomePageView))
