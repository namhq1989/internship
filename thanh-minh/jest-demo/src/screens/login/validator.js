import { AppConst } from '../../configs'
import { KeysConst } from '../../configs/locale'

const PASS_MAX_LENGTH = 6
const validateLogin = {
  isValid: false,
  message: ''
}
/**
 *
 * @param {String} email form ipnut email
 * @param {String} password form input password
 * @returns Object contains: { isValid, message }
 */
const loginValidator = (email, password) => {
  if (!email || !password) {
    validateLogin.message = KeysConst.requireEmailAndPassword
  } else if (!AppConst.regex.email.test(email)) {
    validateLogin.message = KeysConst.emailIsNotValid
  } else if (password.length < PASS_MAX_LENGTH) {
    validateLogin.message = KeysConst.passwordLengthMustAtLeast6Characters
  } else {
    validateLogin.isValid = true
  }
  return validateLogin
}
export {
  loginValidator
}
