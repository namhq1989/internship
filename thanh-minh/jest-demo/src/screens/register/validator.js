import { AppConst } from '../../configs'
import { KeysConst } from '../../configs/locale'

const NAME_MAX_LENGTH = 128
const NAME_MIN_LENGTH = 2
const AGE_MIN = 1
const AGE_MAX = 100
const PASS_MAX_LENGTH = 6

const validate = {
  isValid: false,
  message: ''
}

/**
 *
 * @param {String} name form input name
 * @param {String} email form input email
 * @param {String} password form input password
 * @param {String} rePassword form input confirm password
 * @param {String} age form input age
 * @returns Object validate{ isValid, message}
 */
const registerValidator = (name, email, password, rePassword, age) => {
  if (!name || !email || !password || !rePassword || !age) {
    validate.message = KeysConst.requireRegister
  } else if (!AppConst.regex.email.test(email)) {
    validate.message = KeysConst.emailIsNotValid
  } else if (name.length < NAME_MIN_LENGTH || name.length > NAME_MAX_LENGTH) {
    validate.message = KeysConst.userNameLengthBetweenFrom2To128Characters
  } else if (password.length < PASS_MAX_LENGTH) {
    validate.message = KeysConst.passwordLengthMustAtLeast6Characters
  } else if (rePassword !== password) {
    validate.message = KeysConst.comfirmToFirstPassword
  } else if (!age.match(AppConst.regex.ageInt)) {
    validate.message = KeysConst.ageIntegerNumber
  } else if (age < AGE_MIN || age > AGE_MAX) {
    validate.message = KeysConst.ageMinMax
  } else {
    validate.isValid = true
  }
  return validate
}
export {
  registerValidator
}
