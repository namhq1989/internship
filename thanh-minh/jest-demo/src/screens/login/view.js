import React from 'react'
import { connect } from 'dva'
import { Col, Row, Form, Icon, Button, Input } from 'antd'
import { translate } from 'react-i18next'
import { ImageConst, AppConst } from '../../configs'
import { RcNotification, AccountKitView, LanguageChange } from '../../components'
import { loginValidator } from './validator'
import { KeysConst } from '../../configs/locale'
import './style.less'

const FormItem = Form.Item
export class LoginView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  /**
   *
   * @param {Object} event
   * Handle change form input::
   * Update state on input change ( Email, Password )
   *
  */
  handleChange = (event) => {
    const newState = {}
    newState[event.target.name] = event.target.value
    this.setState(newState)
  }

  /**
   * Submit form login,
   * (check validator) ? (dispatch to models): notification error
   *
  */
  submitLogin = () => {
    const { t, dispatch } = this.props
    const { email, password } = this.state
    const validator = loginValidator(email, password)
    if (validator.isValid) {
      const user = { email, password }
      dispatch({
        type: 'login/login',
        payload: user
      })
    } else {
      RcNotification(t(validator.message), AppConst.notification.error)
    }
  }

  /**
   * Render Login View
   */
  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="wrapper-login">
        <Row type="flex" justify="center">
          <Col xs={18} sm={18} md={8} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="logo" />
            </div>
            <Form className="form-login">
              <FormItem>
                <Input
                  prefix={<Icon type="user" />}
                  id="email"
                  type="email"
                  placeholder={t(KeysConst.emailPalceholder)}
                  name="email"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem>
                <Input
                  prefix={<Icon type="lock" />}
                  id="password"
                  type="password"
                  placeholder={t(KeysConst.passwordPlaceholder)}
                  name="password"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitLogin}
                />
              </FormItem>
              <FormItem className="div-login-button">
                <Button
                  type="primary"
                  htmlType="button"
                  className="button-login"
                  onClick={this.submitLogin}
                >
                  {t(KeysConst.loginButton)}
                </Button>
              </FormItem>
            </Form>
            <AccountKitView />
            <div className="div-login">
              <span>{t(KeysConst.doNotHaveAnAccount)}&nbsp;&nbsp;</span>
              <a href="/#/register" title="register">
                {t(KeysConst.signUp)}
              </a>
            </div>
            <LanguageChange
              t={t}
              changeLanguage={changeLanguage}
            />
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect()(translate('translations')(LoginView))
