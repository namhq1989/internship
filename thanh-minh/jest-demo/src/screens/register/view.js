import React from 'react'
import { connect } from 'dva'
import { Col, Row, Form, Icon, Input, Button } from 'antd'
import { translate } from 'react-i18next'
import { ImageConst, AppConst } from '../../configs'
import { RcNotification, LanguageChange } from '../../components'
import { registerValidator } from './validator'
import { KeysConst } from '../../configs/locale'
import './style.less'


const FormItem = Form.Item
export class RegisterView extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      email: '',
      password: '',
      rePassword: '',
      age: ''
    }
  }

  /**
   * @param {Object} event
   * Handle change form input::
   * Update state on input change (name, email, password, repassword, age)
   */
  handleChange = (event) => {
    // event.preventDefault()
    const newState = {}
    newState[event.target.name] = event.target.value
    this.setState(newState)
  }

  /**
   * Submit Register
   * (check validator) ? (dispatch to models): notification error
   */
  submitRegister = () => {
    const { email, password, rePassword, name, age } = this.state
    const { t, dispatch } = this.props
    const validator = registerValidator(name, email, password, rePassword, age)
    if (validator.isValid) {
      const account = { name, email, password, rePassword, age }
      dispatch({
        type: 'register/register',
        payload: account
      })
    } else {
      RcNotification(t(validator.message), AppConst.notification.error)
    }
  }

  /**
   * Render Register View
   */
  render() {
    const { t, i18n } = this.props
    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng)
    }
    return (
      <div className="wrapper-register">
        <Row type="flex" justify="center">
          <Col xs={18} sm={18} md={8} className="container">
            <div className="logo">
              <img src={ImageConst.imageLogin} alt="logo" />
              <h2 className="title-register">{t(KeysConst.signUp)}</h2>
            </div>
            <Form className="form-register">
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="user" />}
                  type="text"
                  placeholder={t(KeysConst.usernamePlaceholder)}
                  name="name"
                  className="input"
                  id="username"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="mail" />}
                  type="email"
                  placeholder={t(KeysConst.emailPalceholder)}
                  name="email"
                  id="email"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t(KeysConst.passwordPlaceholder)}
                  name="password"
                  id="password"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="lock" />}
                  type="password"
                  placeholder={t(KeysConst.repasswordPlaceholder)}
                  name="rePassword"
                  id="repassword"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="margin-input">
                <Input
                  prefix={<Icon type="idcard" />}
                  type="text"
                  placeholder={t(KeysConst.agePlaceholder)}
                  name="age"
                  id="age"
                  className="input"
                  onChange={this.handleChange}
                  onPressEnter={this.submitRegister}
                />
              </FormItem>
              <FormItem className="div-register-button">
                <Button
                  type="pi18nKeyrimary"
                  htmlType="button"
                  className="button-register"
                  onClick={this.submitRegister}
                >
                  {t(KeysConst.signUp)}
                </Button>
              </FormItem>
            </Form>
            <div className="div-login">
              <span>{t(KeysConst.doHaveAnAccount)}&nbsp;&nbsp;</span>
              <a href="/#/login" title="login">
                {t(KeysConst.loginButton)}
              </a>
            </div>
            <LanguageChange
              t={t}
              changeLanguage={changeLanguage}
            />
          </Col>
        </Row>
      </div>
    )
  }
}
export default connect()(translate('translations')(RegisterView))
