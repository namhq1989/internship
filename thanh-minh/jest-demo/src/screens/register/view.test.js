import React from 'react'
import { shallow } from 'enzyme'
import { RegisterView } from './view'
import { registerValidator } from './validator'

describe('Register View: ', () => {
  const t = jest.fn()
  const handleChange = jest.fn()
  test('render correctly ', () => {
    const wrapper = shallow(
      <RegisterView t={t} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('Update state on input change', () => {
    const wrapper = shallow(
      <RegisterView t={t} handleChange={handleChange} />
    )
    wrapper.find('#username').simulate('change', {
      target: { name: 'name', value: 'Minh123' }
    })
    expect(wrapper.state('name')).toEqual('Minh123')

    wrapper.find('#email').simulate('change', {
      target: { name: 'email', value: 'blah@gmail.com' }
    })
    expect(wrapper.state('email')).toEqual('blah@gmail.com')

    wrapper.find('#password').simulate('change', {
      target: { name: 'password', value: 'minh12345' }
    })
    expect(wrapper.state('password')).toEqual('minh12345')

    wrapper.find('#repassword').simulate('change', {
      target: { name: 'rePassword', value: 'minh12345' }
    })
    expect(wrapper.state('rePassword')).toEqual('minh12345')

    wrapper.find('#age').simulate('change', {
      target: { name: 'age', value: '22' }
    })
    expect(wrapper.state('age')).toEqual('22')
  })

  test('when click button => Call submit login with account user incorrect ', () => {
    const submitRegister = jest.fn()
    const dispatch = jest.fn()
    const wrapper = shallow(
      <RegisterView
        t={t}
        handleChange={handleChange}
        submitRegister={submitRegister}
        dispatch={dispatch}
      />
    )
    wrapper.find('#username').simulate('change', {
      target: { name: 'name', value: 'Minh123' }
    })
    wrapper.find('#email').simulate('change', {
      target: { name: 'email', value: 'blahmail' }
    })
    wrapper.find('#password').simulate('change', {
      target: { name: 'password', value: '126' }
    })
    wrapper.find('#repassword').simulate('change', {
      target: { name: 'rePassword', value: '123456' }
    })
    wrapper.find('#age').simulate('change', {
      target: { name: 'age', value: '22' }
    })

    const user = {
      name: wrapper.state('name'),
      email: wrapper.state('email'),
      password: wrapper.state('password'),
      rePassword: wrapper.state('rePassword'),
      age: wrapper.state('age')
    }
    wrapper.find('.button-register').simulate('click')
    expect(registerValidator(
      user.name,
      user.email,
      user.password,
      user.rePassword,
      user.age
    ).isValid).toEqual(false)
  })

  test('when click button => Call submit login with account user correct ', () => {
    const submitRegister = jest.fn()
    const dispatch = jest.fn()
    const wrapper = shallow(
      <RegisterView
        t={t}
        handleChange={handleChange}
        submitRegister={submitRegister}
        dispatch={dispatch}
      />
    )
    wrapper.find('#username').simulate('change', {
      target: { name: 'name', value: 'Minh123' }
    })
    wrapper.find('#email').simulate('change', {
      target: { name: 'email', value: 'bla@gmail.com' }
    })
    wrapper.find('#password').simulate('change', {
      target: { name: 'password', value: '123456' }
    })
    wrapper.find('#repassword').simulate('change', {
      target: { name: 'rePassword', value: '123456' }
    })
    wrapper.find('#age').simulate('change', {
      target: { name: 'age', value: '22' }
    })

    const user = {
      name: wrapper.state('name'),
      email: wrapper.state('email'),
      password: wrapper.state('password'),
      rePassword: wrapper.state('rePassword'),
      age: wrapper.state('age')
    }
    wrapper.find('.button-register').simulate('click')
    expect(registerValidator(
      user.name,
      user.email,
      user.password,
      user.rePassword,
      user.age
    ).isValid).toEqual(true)
  })
})
