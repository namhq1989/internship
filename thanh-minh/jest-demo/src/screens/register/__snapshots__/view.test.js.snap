// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register View:  render correctly  1`] = `
<div
  className="wrapper-register"
>
  <Row
    gutter={0}
    justify="center"
    type="flex"
  >
    <Col
      className="container"
      md={8}
      sm={18}
      xs={18}
    >
      <div
        className="logo"
      >
        <img
          alt="logo"
          src={null}
        />
        <h2
          className="title-register"
        />
      </div>
      <Form
        className="form-register"
        hideRequiredMark={false}
        layout="horizontal"
        onSubmit={[Function]}
        prefixCls="ant-form"
      >
        <FormItem
          className="margin-input"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="username"
            name="name"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="user"
              />
            }
            prefixCls="ant-input"
            type="text"
          />
        </FormItem>
        <FormItem
          className="margin-input"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="email"
            name="email"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="mail"
              />
            }
            prefixCls="ant-input"
            type="email"
          />
        </FormItem>
        <FormItem
          className="margin-input"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="password"
            name="password"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="lock"
              />
            }
            prefixCls="ant-input"
            type="password"
          />
        </FormItem>
        <FormItem
          className="margin-input"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="repassword"
            name="rePassword"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="lock"
              />
            }
            prefixCls="ant-input"
            type="password"
          />
        </FormItem>
        <FormItem
          className="margin-input"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Input
            className="input"
            disabled={false}
            id="age"
            name="age"
            onChange={[Function]}
            onPressEnter={[Function]}
            prefix={
              <Icon
                type="idcard"
              />
            }
            prefixCls="ant-input"
            type="text"
          />
        </FormItem>
        <FormItem
          className="div-register-button"
          colon={true}
          hasFeedback={false}
          prefixCls="ant-form"
        >
          <Button
            className="button-register"
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
            prefixCls="ant-btn"
            type="pi18nKeyrimary"
          />
        </FormItem>
      </Form>
      <div
        className="div-login"
      >
        <span>
            
        </span>
        <a
          href="/#/login"
          title="login"
        />
      </div>
      <LanguageChange
        changeLanguage={[Function]}
        t={
          [MockFunction] {
            "calls": Array [
              Array [
                "signUp",
              ],
              Array [
                "usernamePlaceholder",
              ],
              Array [
                "emailPalceholder",
              ],
              Array [
                "passwordPlaceholder",
              ],
              Array [
                "repasswordPlaceholder",
              ],
              Array [
                "agePlaceholder",
              ],
              Array [
                "signUp",
              ],
              Array [
                "doHaveAnAccount",
              ],
              Array [
                "loginButton",
              ],
            ],
          }
        }
      />
    </Col>
  </Row>
</div>
`;
